### PR TITLE
fix: resolve open P2 issues

### PR DESCRIPTION
**Pull request by XiNian-dada**
## Summary
- Fix all currently open P2 issues: #237, #236, #234, #232, #231, #208, #201, #162.
- Add protocol fuzz smoke coverage, split oversized registry/macOS collector modules, harden registry/web asset edge cases, accept documented TOTP drift, extract the UI dedupe composable, and decouple server registry runtime state from external response models.

## Verification
- cargo fmt --check
- cargo test --workspace
- cargo clippy --all-targets -- -D warnings
- pnpm test
- pnpm typecheck
- pnpm lint
- pnpm build
- cargo test --manifest-path fuzz/Cargo.toml
- cargo run --manifest-path fuzz/Cargo.toml --bin protocol_messages -- 10000
- cargo run --manifest-path fuzz/Cargo.toml --bin wire_message -- fuzz/corpus/protocol_messages
- cargo run --manifest-path fuzz/Cargo.toml --bin browser_message -- fuzz/corpus/protocol_messages

### CODE REVIEW

File: nodelite-server/src/auth.rs [hunk 1]
```diff
@@ -378,8 +378,19 @@ fn verify_totp_step_at(totp_secret: Option<&[u8]>, code: &str, now_step: u64) ->
         return None;
     }
 
-    let expected = totp_code_for_step(secret, now_step);
-    constant_time_compare_bytes(expected.as_bytes(), code.as_bytes()).then_some(now_step)
+    let candidate_steps = now_step
```
Comment (XiNian-dada) [inline nodelite-server/src/auth.rs:382]: [P2] This returns the first matching step in previous/current/next order, but callers only replay-check that single returned step. Adjacent 6-digit TOTP steps can collide; for example, with the RFC test secret `12345678901234567890`, steps `910737` and `910738` both generate `911617`. If the previous step was already consumed, the current step's valid same code will be rejected as replay instead of considering the current unused match. Please return all matching candidate steps, or pass an `is_unused(step)` predicate into verification and select a matching unused step, then add a regression test for adjacent-step collision.

**Review by XiNian-dada — COMMENTED** [review]
Reviewed PR #241. I left one P2 inline finding on the TOTP drift/replay behavior.

File: nodelite-server/src/state/registry.rs [hunk 2]
```diff
@@ -292,48 +380,46 @@ impl Registry {
...
-                },
-            );
+            self.nodes
+                .insert(node_id.clone(), NodeEntry::from_restored_status(status));
             self.sorted_node_ids.push(node_id);
         }
         self.resort_node_ids();
     }
 
+    #[cfg(test)]
```
Comment (XiNian-dada) [inline nodelite-server/src/state/registry.rs:402]: [P2] This does not yet satisfy #162's acceptance criteria. `size_of::<NodeEntry>()` only compares inline struct layout; it does not show RSS, allocation count, clone cost, or CPU improvement under the large snapshot / many-disks workload that motivated the issue. `NodeEntry` still stores a full `NodeSnapshot` and reconstructs/clones full `NodeStatus` for detail responses, so the PR should include a reproducible before/after measurement for at least one of RSS, CPU, or allocations, or stop claiming #162 is fully resolved.

**Review by XiNian-dada — COMMENTED** [review]
Adding one more review note for the #162 performance acceptance gap.

**Comment by XiNian-dada** [comment]
Follow-up fixes pushed in `d3bf3d6` and `f39b9d3`:

- TOTP drift verification now evaluates all matching adjacent steps before replay rejection, and marks all matching steps as consumed after acceptance. Added a regression for adjacent-step 6-digit collisions.
- #162 now has a reproducible retained-heap estimate test using a large node snapshot / many disks workload. It compares retained heap bytes and heap buffer counts for the runtime entry model against the previous cached external models, instead of relying only on inline `size_of`.

Local verification passed:

- `cargo +stable fmt --all --check`
- `NODELITE_SKIP_WEB_BUILD=1 cargo test --workspace`
- `NODELITE_SKIP_WEB_BUILD=1 cargo clippy --all-targets -- -D warnings`
- `pnpm test`
- `pnpm typecheck`
- `pnpm lint`
- `pnpm build`

Remote GitHub Actions are still running; I am waiting for them before merging.

**Comment by XiNian-dada** [comment]
Remote verification completed before merge:

- CI: success
- Coverage: success
- All review threads resolved

Merged into `main` as `3c1406907b8f7746d4773fd60b80ceed498afab5`.
